extension/src/popup/components/sendPayment/SendTo/index.tsx

```diff
@@ -29,33 +35,49 @@ export const SendTo = () => {
 
   const dispatch = useDispatch();
   const [isLoading, setIsLoading] = useState(false);
-  const [isValidAddress, setIsValidAddress] = useState(false);
   const [destinationBalances, setDestinationBalances] = useState(
     defaultAccountBalances,
   );
   const networkDetails = useSelector(settingsNetworkDetailsSelector);
 
-  const validatePublicKey = (publicKey: string) => {
+  const handleContinue = (values: { destination: string }) => {
+    dispatch(saveDestination(values.destination));
+    formik.resetForm({ values });
+    navigateTo(ROUTES.sendPaymentAmount);
+  };
+
+  const formik = useFormik({
+    initialValues: { destination },
+    onSubmit: handleContinue,
+    validateOnChange: false,
+    validate: (values) => {
+      if (validPublicKey(values.destination)) {
+        return {};
+      }
+      return { destination: "invalid public key" };
+    },
+  });
+
+  // TODO - handle federation address and muxed accounts
+  const validPublicKey = (publicKey: string) => {
```

Comment:
NIT: since this returns a boolean, can you rename to `isValidPublicKey`? It's a naming convention we try to use for boolean values

https://github.com/stellar/product-conventions/blob/master/STYLEGUIDE.md#booleans-should-be-named-with-is-has-did-etc

Reply:
👍  good call https://github.com/stellar/freighter/pull/381/commits/b6387b726fe3bfebc60f8ed016592b4a21d86e35